gen/quick_test.go

```diff
@@ -148,6 +148,34 @@ func enumValueGenerator(valuesFunc interface{}) func(*testing.T, *rand.Rand) thr
 	}
 }
 
+// copyWithDefaults copies the given thrift object to a new object
+// override nil values from the `give` parameter with defaults when
+// available.
+func copyWithDefaults(give, defaults thriftType) thriftType {
+	if defaults == nil {
+		return deepCopy(give)
+	}
+	d := reflect.ValueOf(defaults).Elem()
+	g := reflect.ValueOf(give).Elem()
+
+	// copy values into a new struct, selecting default values from
+	// d when g values are nil
+	w := reflect.New(reflect.TypeOf(give).Elem()).Elem()
```

Comment:
Can probably simplify to
```suggestion
	w := reflect.New(g.Type()).Elem()
```